diff -Naur xpra.orig/xpra-5.0.8/xpra/x11/vfb_util.py xpra/xpra-5.0.8/xpra/x11/vfb_util.py
--- xpra.orig/xpra-5.0.8/xpra/x11/vfb_util.py   2024-04-03 17:14:38.000000000 +0200
+++ xpra/xpra-5.0.8/xpra/x11/vfb_util.py        2024-06-12 23:51:38.914626775 +0200
@@ -282,6 +282,9 @@
     try:
         if use_display_fd:
             def displayfd_err(msg):
+                if xvfb and xvfb.poll() is None:
+                    log.error(" stopping vfb process with pid %i", xvfb.pid)
+                    xvfb.terminate()
                 raise InitException(f"{xvfb_executable}: {msg}")
             r_pipe, w_pipe = os.pipe()
             try:
