diff --git a/bin/build_flang_runtime.sh b/bin/build_flang_runtime.sh
index 6e82b6bc..fe7e7ca1 100755
--- a/bin/build_flang_runtime.sh
+++ b/bin/build_flang_runtime.sh
@@ -122,7 +122,8 @@ fi
 cd $BUILD_DIR/build/flang_runtime
 
 #  Need llvm-config to come from previous LLVM build
-export PATH=$AOMP_INSTALL_DIR/bin:$PATH
+#  Need to be able to find flang1
+export PATH=$AOMP_INSTALL_DIR/bin:$BUILD_DIR/build/flang/bin:$PATH
 # Old CMAKE uses FC env variable to find fortran compiler
 export FC=$AOMP_INSTALL_DIR/bin/flang
 
