diff --git a/bin/build_openmp.sh b/bin/build_openmp.sh
index 4ef95075..526abb89 100755
--- a/bin/build_openmp.sh
+++ b/bin/build_openmp.sh
@@ -151,7 +151,7 @@ if [ "$1" != "nocmake" ] && [ "$1" != "install" ] ; then
          env "$@" ${AOMP_CMAKE} $MYCMAKEOPTS  $BUILD_DIR/$AOMP_PROJECT_REPO_NAME/openmp
       else
          echo ${AOMP_CMAKE} $MYCMAKEOPTS  $AOMP_REPOS/$AOMP_PROJECT_REPO_NAME/openmp
-         env "$@" ${AOMP_CMAKE} $MYCMAKEOPTS  $AOMP_REPOS/../$AOMP_PROJECT_REPO_NAME/openmp
+         env "$@" ${AOMP_CMAKE} $MYCMAKEOPTS  $AOMP_REPOS/$AOMP_PROJECT_REPO_NAME/openmp
       fi
       if [ $? != 0 ] ; then 
          echo "ERROR openmp cmake failed. Cmake flags"
@@ -187,8 +187,8 @@ if [ "$1" != "nocmake" ] && [ "$1" != "install" ] ; then
          env "$@" ${AOMP_CMAKE} $MYCMAKEOPTS  $BUILD_DIR/$AOMP_PROJECT_REPO_NAME/openmp
       else
          # FIXME: Remove CMAKE_CXX_FLAGS and CMAKE_C_FLAGS when AFAR uses 5.3 ROCr.
-         echo ${AOMP_CMAKE} $MYCMAKEOPTS -DCMAKE_CXX_FLAGS="-g -I$ROCM_DIR/include" -DCMAKE_C_FLAGS="-g -I$ROCM_DIR/include" $AOMP_REPOS/../$AOMP_PROJECT_REPO_NAME/openmp
-         env "$@" ${AOMP_CMAKE} $MYCMAKEOPTS -DCMAKE_CXX_FLAGS="-g -I$ROCM_DIR/include" -DCMAKE_C_FLAGS="-g -I$ROCM_DIR/include" $AOMP_REPOS/../$AOMP_PROJECT_REPO_NAME/openmp
+         echo ${AOMP_CMAKE} $MYCMAKEOPTS -DCMAKE_CXX_FLAGS="-g -I$ROCM_DIR/include" -DCMAKE_C_FLAGS="-g -I$ROCM_DIR/include" $AOMP_REPOS/$AOMP_PROJECT_REPO_NAME/openmp
+         env "$@" ${AOMP_CMAKE} $MYCMAKEOPTS -DCMAKE_CXX_FLAGS="-g -I$ROCM_DIR/include" -DCMAKE_C_FLAGS="-g -I$ROCM_DIR/include" $AOMP_REPOS/$AOMP_PROJECT_REPO_NAME/openmp
       fi
       if [ $? != 0 ] ; then 
          echo "ERROR openmp debug cmake failed. Cmake flags"
