diff --git a/src/gemm_v2.cpp b/src/gemm_v2.cpp
index fc8b3222c..f95337dc0 100644
--- a/src/gemm_v2.cpp
+++ b/src/gemm_v2.cpp
@@ -40,7 +40,7 @@
 
 #if MIOPEN_USE_ROCBLAS
 #include <half.hpp>
-#include <rocblas.h>
+#include <rocblas/rocblas.h>
 #include <miopen/perf_field.hpp>
 #endif
 
diff --git a/src/include/miopen/handle.hpp b/src/include/miopen/handle.hpp
index ee45ae2a0..b99883f5f 100644
--- a/src/include/miopen/handle.hpp
+++ b/src/include/miopen/handle.hpp
@@ -52,7 +52,7 @@
 
 #if MIOPEN_USE_ROCBLAS
 #include <miopen/manage_ptr.hpp>
-#include <rocblas.h>
+#include <rocblas/rocblas.h>
 #endif
 
 namespace miopen {
