diff --git a/src/comgr.cpp b/src/comgr.cpp
index cf108f247..0a1d3791f 100644
--- a/src/comgr.cpp
+++ b/src/comgr.cpp
@@ -119,7 +119,7 @@ MIOPEN_DECLARE_ENV_VAR(MIOPEN_DEBUG_OPENCL_WAVE64_NOWGP)
 
 // '__hipGetPCH' is not available in [4.4, 5.0). See SWDEV-308265.
 #if HIP_SUPPORTS_PCH && (HIP_PACKAGE_VERSION_FLAT >= 4004000000ULL) && \
-    (HIP_PACKAGE_VERSION_FLAT < 5000000000ULL)
+    (HIP_PACKAGE_VERSION_FLAT < 5003000000ULL)
 #undef HIP_SUPPORTS_PCH
 #define HIP_SUPPORTS_PCH 0
 #endif
