diff --git a/src/base/hip/backend_hip.cpp b/src/base/hip/backend_hip.cpp
index 72424a9..082cdb1 100644
--- a/src/base/hip/backend_hip.cpp
+++ b/src/base/hip/backend_hip.cpp
@@ -40,7 +40,7 @@
 #include "hip_vector.hpp"
 
 #include <hip/hip_runtime_api.h>
-#include <rocblas.h>
+#include <rocblas/rocblas.h>
 #include <rocsparse.h>
 
 #include <complex>
diff --git a/src/base/hip/hip_blas.cpp b/src/base/hip/hip_blas.cpp
index db67352..9979d16 100644
--- a/src/base/hip/hip_blas.cpp
+++ b/src/base/hip/hip_blas.cpp
@@ -27,7 +27,7 @@
 
 #include <hip/hip_complex.h>
 #include <hip/hip_runtime_api.h>
-#include <rocblas.h>
+#include <rocblas/rocblas.h>
 #include <rocprim/rocprim.hpp>
 
 #include <complex>
diff --git a/src/base/hip/hip_blas.hpp b/src/base/hip/hip_blas.hpp
index 002a796..5a0afc5 100644
--- a/src/base/hip/hip_blas.hpp
+++ b/src/base/hip/hip_blas.hpp
@@ -25,7 +25,7 @@
 #define ROCALUTION_HIP_HIP_BLAS_HPP_
 
 #include <hip/hip_runtime_api.h>
-#include <rocblas.h>
+#include <rocblas/rocblas.h>
 
 namespace rocalution
 {
diff --git a/src/base/hip/hip_conversion.hpp b/src/base/hip/hip_conversion.hpp
index 8c3fa3f..c67a7c8 100644
--- a/src/base/hip/hip_conversion.hpp
+++ b/src/base/hip/hip_conversion.hpp
@@ -27,7 +27,7 @@
 #include "../backend_manager.hpp"
 #include "../matrix_formats.hpp"
 
-#include <rocblas.h>
+#include <rocblas/rocblas.h>
 #include <rocsparse.h>
 
 namespace rocalution
diff --git a/src/base/hip/hip_utils.hpp b/src/base/hip/hip_utils.hpp
index 550beb7..3c3b68b 100644
--- a/src/base/hip/hip_utils.hpp
+++ b/src/base/hip/hip_utils.hpp
@@ -29,7 +29,7 @@
 #include "backend_hip.hpp"
 
 #include <hip/hip_runtime.h>
-#include <rocblas.h>
+#include <rocblas/rocblas.h>
 #include <rocsparse.h>
 
 #ifdef SUPPORT_COMPLEX
