diff --git a/library/src/CMakeLists.txt b/library/src/CMakeLists.txt
index 79f54f21..ace140f6 100755
--- a/library/src/CMakeLists.txt
+++ b/library/src/CMakeLists.txt
@@ -493,6 +493,11 @@ include( GenerateExportHeader )
 set_target_properties( rocblas PROPERTIES CXX_VISIBILITY_PRESET "hidden" C_VISIBILITY_PRESET "hidden" VISIBILITY_INLINES_HIDDEN ON )
 generate_export_header( rocblas EXPORT_FILE_NAME ${PROJECT_BINARY_DIR}/include/rocblas/internal/rocblas-export.h )
 
+if (NOT python)
+    find_package(Python COMPONENTS Interpreter)
+    set(python ${Python_EXECUTABLE})
+endif()
+
 # generate header with prototypes for export reuse
 file( GLOB rocblas_prototype_inputs
   LIST_DIRECTORIES OFF
