diff --git a/library/src/include/envariables.h b/library/src/include/envariables.h
index b794b23b..9b79bd0f 100644
--- a/library/src/include/envariables.h
+++ b/library/src/include/envariables.h
@@ -24,6 +24,8 @@
 
 #pragma once
 
+#include <cstdint>
+
 template <std::size_t N, typename T>
 inline constexpr std::size_t countof(T (&)[N])
 {
@@ -72,8 +74,8 @@ public:
     //
     // Number of values.
     //
-    static constexpr size_t size          = countof(all);
-    static constexpr size_t bool_var_size = size;
+    static constexpr std::size_t size          = countof(all);
+    static constexpr std::size_t bool_var_size = size;
 
     //
     // \brief Return value of a Boolean variable.
