diff --git a/CMakeLists.txt b/CMakeLists.txt
index b04de88..0dfacb0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,10 +110,10 @@ endif()
 
 set(HIPIFY_INSTALL_PATH ${CMAKE_INSTALL_PREFIX})
 
-install(TARGETS hipify-clang DESTINATION ${CMAKE_INSTALL_PREFIX})
+install(TARGETS hipify-clang DESTINATION ${CMAKE_INSTALL_PREFIX}/bin/)
 install(
     DIRECTORY ${CMAKE_SOURCE_DIR}/bin/
-    DESTINATION ${CMAKE_INSTALL_PREFIX}
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/bin/
     USE_SOURCE_PERMISSIONS
     PATTERN "hipify-perl"
     PATTERN "*.sh")
