--- setup_orig.py	2021-01-20 10:10:11.609352000 +0100
+++ setup.py	2021-01-20 10:10:37.525550350 +0100
@@ -39,7 +39,7 @@
         define_macros += [('WITH_CUDA', None)]
         nvcc_flags = os.getenv('NVCC_FLAGS', '')
         nvcc_flags = [] if nvcc_flags == '' else nvcc_flags.split(' ')
-        nvcc_flags += ['-arch=sm_35', '--expt-relaxed-constexpr']
+        nvcc_flags += ['-gencode=arch=compute_80,code=sm_80', '--expt-relaxed-constexpr']
         extra_compile_args['nvcc'] = nvcc_flags
 
     extensions_dir = osp.join(osp.dirname(osp.abspath(__file__)), 'csrc')
